templates/typescript_gapic/src/$version/$service_client.ts.njk

```diff
@@ -49,42 +50,56 @@ export class {{ service.name }}Client {
 
   constructor(opts: ClientOptions) {
     // Ensure that options include the service address and port.
+    opts = opts || {};
+    const isBrowser = (typeof window !== 'undefined');
+    if (isBrowser){
+      opts.fallback = true;
+    }
+    // If we are in browser, we are already using fallback because of the
+    // "browser" field in package.json.
+    // But if we were explicitly requested to use fallback, let's do it now.
+    const gaxModule = !isBrowser && opts.fallback ? gax.fallback : gax;
+    const servicePath =
+      opts.servicePath ||
+      opts.apiEndpoint ||
+      (this.constructor as typeof {{service.name}}Client).servicePath;
+
     opts = Object.assign(
         {
           clientConfig: {},
           port: (this.constructor as typeof {{ service.name }}Client).port,
-          servicePath: (this.constructor as typeof {{ service.name }}Client).servicePath,
+          servicePath: servicePath,
         },
         opts);
 
     // Create a `gaxGrpc` object, with any grpc-specific options
     // sent to the client.
     opts.scopes = (this.constructor as typeof {{ service.name }}Client).scopes;
-    const gaxGrpc = new gax.GrpcClient(opts);
+    const gaxGrpc = new gaxModule.GrpcClient(opts);
 
     // Save the auth object to the client, for use by other methods.
-    this.auth = gaxGrpc.auth;
+    this.auth = (gaxGrpc.auth as gax.GoogleAuth);
 
     // Determine the client header string.
     const clientHeader = [
       `gl-node/${process.version}`,
       `grpc/${gaxGrpc.grpcVersion}`,
-      `gax/${gax.version}`,
+      `gax/${gaxModule.version}`,
       `gapic/${version}`,
+      `gl-web/${gaxModule.version}`
     ];
     if (opts.libName && opts.libVersion) {
       clientHeader.push(`${opts.libName}/${opts.libVersion}`);
     }
+    // Load the applicable protos.
+    // For Node.js, pass the path to JSON proto file.
+    // For browsers, pass the JSON content.
 
-    const gaxProtos = Object.assign(
-        {},
-{%- set protosJoiner = joiner() %}
-{%- for filename in api.filesToGenerate %}
-      {{- protosJoiner() }}
-        gaxGrpc.loadProto(
-            path.join(__dirname, '..', '..', 'protos'),
-            '{{ filename }}')
-{%- endfor -%}
+    const typescriptProtoPath = path.join(__dirname, '..', '..', 'protos', 'protos.json');
```

Comment:
let's use the same variable name as in current libraries (`nodejsProtoPath` if I remember correctly) to minimize differences